pkg/controller/controller.go

```diff
@@ -119,7 +119,8 @@ func (c *AppGwIngressController) ProcessEvent(event events.Event) error {
 		glog.Error("Error mutating AKS from k8s event. ", err)
 	}
 
-	if err := c.MutateAppGateway(appGw, cbCtx); err != nil {
+	invokedForReconciliation := event.Type == events.PeriodicReconcile
+	if err := c.MutateAppGateway(appGw, cbCtx, invokedForReconciliation); err != nil {
```

Comment:
Decision to whether mutate appgw due to any event should be togather. 
I think its better if we move out the logic of reconcile out of here

Should Provess Event should check for reconcileEvent, 
It should return false if any event has happened in time less than reconcile time. 
If it is decided to process that event, then enter ProcessEvent.

Process Event should not have logic of any specific event type

Reply:
sure, I can move the reconcile logic to ShouldProcess.